### ISSUE TITLE
PropertyInfo XAML control not updating after bound property becomes valid

### ISSUE THREAD

**Issue by benstickel** [MarimerLLC/cslaforum#718]
**Question**

I'm working on a Xamarin.Forms project with CSLA 4.8.1. I have some Csla.XAML.PropertyInfo controls on my form that should be dynamically changing some label's text colors depending on the IsValid state of the associated properties. My model class has two child properties, one is a MobileList<elementEdit> and one is an elementEdit that once validated gets added to the list. If the model object is new and the list is empty then the list property should be marked as invalid, resulting in a red label. 

In the initial state, the list property is correctly marked invalid and reflected that way in the label color.

The first business rule in my model has the model's elementEdit to add property as the primary property and during its execution it is successfully adding the element to my list via context.AddOutValue(). My list property is being included as an input property as well as an affected property so that the validation rules for the list property run after this rule.

The second rule is a validation rule with the primary property being the list property. This rule is successfully updating the model's BrokenRulesCollection to be empty now that the list is no longer empty due to the business rule's execution. 

The issue is that the Csla.XAML.PropertyInfo control's BrokenRulesCollection becomes out of sync at this point, still showing a broken business rule for my list property when in fact the list property is valid. 

I can remedy the issue for now by clearing the Path property on the control and resetting it to the list property of my model but this feels like a hack and I wouldn't be surprised if I am incorrectly implementing the business rules. I suspect the PropertyHasChanged event for the list property is not being handled by the Csla.XAML.PropertyInfo control as I'd expect because the validation rule for the list property is running as an affected property after the elementEdit property's business rule finishes execution.

Is there a way to set up my rules so that my label stays in sync with the associated property's IsValid state? Should I be leveraging rule-chaining?

**Version and Platform**
CSLA version: 4.8.100
OS: Android
Platform: Xamarin

**Comment by rockfordlhotka** [comment]
You are attaching the PropertyInfo control to the list, not to a specific child object?

**Comment by benstickel** [comment]
The MobileList object is a child property on my model. My PropertyInfo control is attached to the model's child property.

**Comment by rockfordlhotka** [comment]
Yeah, so the PropertyInfo control is attached to a list object, not a singular property.

Sadly right now that's not a supported scenario - hence the work item.

**Comment by benstickel** [comment]
Thank you for your quick response nonetheless. It's nice knowing it wasn't a mistake on my part.
